### PR TITLE
New eslint rules

### PR DESCRIPTION
**Pull request by pittst3r**
## Motivation

Our current eslint config is unopinionated about whitespace and semicolons, and I am. Seeing diffs that add whitespace at the end of lines or the removal of a newline character at the end of a file drives me a little nuts. Just a little bit.

## Approach

- Sort the rules alphabetically
- Add the following rules:
  - `"@typescript-eslint/no-empty-function": "off"`: No-op functions are fairly common and in my experience this rule does not add value, only annoyance due to needing to put a `return` or a comment in no-ops.
  - `"arrow-spacing": "error"`: Adds whitespace around arrows for tidy, consistent formatting.
  - `"eol-last": "error"`: I'm not sure if this is technically required still, but having a newline at the end of a file is at least traditional and github complains at you in the diff view if you don't include a newline.
  - `"no-empty-function": "off"`: This is necessary for `"@typescript-eslint/no-empty-function": "off"` to work.
  - `"no-multi-spaces": "error"`: Eliminates multiple consecutive spaces where you only need one for the sake of tidy, consistent formatting.
  - `"no-trailing-spaces": "error"`: Eliminates trailing spaces at the end of a line. This is for tidiness as well, and it's very minor because you can only see it in a diff, but it's also auto-fixable, so I was inclined to include it.
  - `"object-curly-spacing": ["error", "always"]`: Adds whitespace padding inside object curlies for tidiness and consistency. It's easier on the eyes to parse objects when they have this whitespace, especially if there are multiple levels of curlies.
  - `"one-var": ["error", { initialized: "never" }]`: This allows `let foo, bar;` but disallows `let foo = 1, bar = 2;`. These initializations are usually spread across multiple lines and can be easily confused with a variable reassignment if you miss the `let` or `const` multiple lines up.
  - `semi: ["error", "always", { omitLastInOneLineBlock: true }]`: Enforces a semicolon usage, but allows one-line blocks to omit the semicolon (`element => { element.click() }`). This is for consistency and tidiness. I am not dogmatic about semis vs. no-semis and would be (almost) as happy enforcing no semis.

### CODE REVIEW

**Review by taras — CHANGES_REQUESTED** [review]
It would be helpful if you could include 1 or 2 sentence for each rule to explain why you think we should add them.

**Comment by cowboyd** [comment]
> Add rule disallowing the initialization of variables using the comma syntax (like let foo = 1, bar = 2;)

Agree whole-heartedly, although going to date myself by recalling when this was actually the preferred nodejs style:

```js
var x = 'one'
  , y = 'two'
  , z = 'three';
```

🤣 😆 😭

**Comment by cowboyd** [comment]
> It would be helpful if you could include 1 or 2 sentence for each rule to explain why you think we should add them.

Maybe even to the README, that way, it's front and center to anybody viewing the GH project.

**Comment by pittst3r** [comment]
@taras @cowboyd Thanks for looking. I just updated the description to add detail around my reason for each rule. I agree that it would be good to add reasoning to the readme for each rule. I will add that once we agree on these rule additions.

**Review by taras — APPROVED** [review]
👍 thank you for updating the description

**Comment by cowboyd** [comment]
> No-op functions are fairly common and in my experience this rule does not add value, only annoyance due to needing to put a return or a comment in no-ops.

I'd even add that empty functions are like `0` for functional programming and represent the base-case where the context is absence of computation. You wouldn't want to do math without being allowed to write `0` and instead having to write `(1 - 1)`  everywhere!

**Comment by pittst3r** [comment]
> Agree whole-heartedly, although going to date myself by recalling when this was actually the preferred nodejs style

There was like a two-day period about 6 years ago when I learned of that way of doing commas in objects and arrays and thought it was awesome. 😆

**Comment by pittst3r** [comment]
@cowboyd No rush but I'm going to wait to merge until I have your blessing on this as well.

The Blessing

**Comment by pittst3r** [comment]
Oh just kidding, I don't have write access to this repo. Someone else will need to merge.

**Comment by chuckcarpenter** [comment]
@pittst3r it's like you don't even work here.

**Comment by taras** [comment]
@pittst3r try it now

**Review by cowboyd — APPROVED** [review]
I'm in!

**Pull request merged**
